### PR TITLE
The user is able to delete habits

### PR DESCRIPTION
**Pull request by farhaannishtar**

### CODE REVIEW

File: components/Habit.js [hunk 1]
```diff
@@ -4,19 +4,30 @@ import InlineEdit from './InlineEdit';
...
+  }
+  const selectCard = () => {
     if (checkmark === "/redCheckMark.svg") {
       setCheckmark("/grayCheckMark.svg");
     } else {
       setCheckmark("/redCheckMark.svg");
     }
+    if (props.habit.selected === false) {
+      setCompletedHabits(completedHabits => completedHabits + 1)
+      props.habit.selected = true;
```
Comment (ahsojar) [inline components/Habit.js]: Updating a prop value within the component is a React antipattern. React props are meant to be **immutable** values passed in by the parent into a child, so its not meant to be updated within a React component like this.

File: components/Habit.js [hunk 2]
```diff
@@ -36,26 +47,31 @@ export default function Habit(props) {
...
-                        {
-                          emoji === "" ? <p className="addEmojiText">☻ Add icon</p> : <div className="emoji">{ emoji }</div>
-                        }
-                        <div className='shadow'></div>
-                      </div>
-                    )
-                  }
+              <div className='topLeft'>
+                <div className='redX'>
+                  <img src={'/redX.svg'} alt="SVG as an image" onClick={deleteHandler}/>
```
Comment (ahsojar) [inline components/Habit.js]: both the delete SVG and the current placement don't look good. Can you point me to the original Dribble you were trying to follow? We need to find something that matches more of the theme you have.

File: pages/index.js [hunk 3]
```diff
@@ -89,18 +86,25 @@ export default function Home({ isConnected }) {
                 id={habit.id} 
                 isEditable={true} 
                 color={cardColors[index % cardColors.length]} 
-                habit={habit} setHabits={setHabits} 
+                habit={habit} 
+                setHabits={setHabits}
+                completedHabits={completedHabits}
```
Comment (ahsojar) [inline pages/index.js]: Why does each single habit component need to know the total completed habits here?

Comment (ahsojar) [reply pages/index.js]: Ah i see, it looks like you're using it in the child to update the count of completed habits. You could potentially update the parent compnent here to

` onComplete={() => setCompletedHabits(completedHabits +1) }`

which would let you delete the logic you're doing in the individual Habit component. Ideally the Habit component is "dumb" and doesn't need to care about the outside world / maintaining state of the entire app, if that makes sense.  It should just need to worry about calling whatever `onComplete` that's passed down, without needing to worry about the actual logic.

**Review by ahsojar — CHANGES_REQUESTED** [review]

File: components/Habit.js (hunk 2 above)
Comment (farhaannishtar) [reply components/Habit.js]: https://dribbble.com/shots/18932568-Morin-Habit-Tracker-App
Here's my dribble ^

In the dribble I followed, there was no button to delete habits.

File: components/Habit.js [hunk 4]
```diff
@@ -8,15 +8,28 @@ export default function Habit(props) {
...
+    if (props.habit.completed === false) {
+      props.updateCompletedHabits(props.id, true);
+    } else {
+      props.updateCompletedHabits(props.id, false);
+    }
+    if (textDecoration === 'none') {
+      setTextDecoration('line-through');
+    } else {
+      setTextDecoration('none')
+    }
```
Comment (ahsojar) [inline components/Habit.js]: This whole function would be a lot easier to read if the logic was grouped based on whether we are setting things to true or false, example:

```
const completeHabit = () => {
    if (props.habit.completed === false) {
      props.updateCompletedHabits(props.id, true);
      setCheckmark("/grayCheckMark.svg");
      setTextDecoration('line-through');
    } else {
     ....
    }
   ```

Doing it this way lets you 1) reduce the number of if/elses 2) Makes it easier to see what's happening in when a user does some action.

File: pages/index.js [hunk 5]
```diff
@@ -66,15 +66,31 @@ export default function Home({ isConnected }) {
...
+    }))
+  }
+
+  useEffect(() => {
+    let count = 0;
+    habits.forEach(habit => {
+      if (habit.completed === true) count++;
+    })
+    setCompletedHabits(count);
+  }, [habits])
```
Comment (ahsojar) [inline pages/index.js]: Do you need a variable to keep track of a completed habits/have this useEffect at all? It seems like you could infer this by just looking at your `habits` array and using something like [filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to get the # of completed habits.

File: components/Habit.js [hunk 6]
```diff
@@ -36,35 +49,23 @@ export default function Habit(props) {
...
                 </div>
               </label>
             </div> 
-            <div onClick={(e) => e.stopPropagation()}>
-            <InlineEdit editingValue={editingValue} setEditingValue={setEditingValue}/>
+            <div onClick={(e) => e.stopPropagation()} className='inline'>
+              <InlineEdit editingValue={editingValue} setEditingValue={setEditingValue} textDecoration={textDecoration}/>
+            </div>
+            <div className='trashcan'>
+              <img src='/trash.svg' onClick={deleteHandler}/>
```
Comment (ahsojar) [inline components/Habit.js:62]: Can we move the trashcan lower on the card?
CleanShot 2023-01-09 at 19 52 58@2x

File: components/Habit.js (hunk 4 above)
Comment (farhaannishtar) [reply components/Habit.js]: Done

File: pages/index.js (hunk 5 above)
Comment (farhaannishtar) [reply pages/index.js]: Done. I just had to type, `habits.reduce` and github copilot wrote `habits.reduce((acc, habit) => acc + (habit.completed ? 1 : 0), 0)` (I understand the code)

File: components/Habit.js (hunk 6 above)
Comment (farhaannishtar) [reply components/Habit.js:62]: Done

**Pull request merged**